spec/services/vine_voucher_validator_service_spec.rb

```diff
@@ -0,0 +1,365 @@
+        allow(vine_api_service).to receive(:voucher_validation).and_return(
+          mock_api_response(success: false, status: 404, data: )
+        )
+      end
+
+      it "returns nil" do
+        expect(validate_voucher_service.validate).to be_nil
+      end
+
+      it "adds an error message" do
+        validate_voucher_service.validate
+
+        expect(validate_voucher_service.errors).to include(
+          { not_found_voucher: "The voucher doesn't exist" }
+        )
+      end
+
+      it "doesn't creates a new VINE voucher" do
+        expect { validate_voucher_service.validate }.not_to change { Voucher.count }
+      end
+    end
+
+    context "when the voucher is an invalid voucher" do
+      let!(:vine_connected_app) {
+        ConnectedApps::Vine.create(
+          enterprise: distributor, data: { api_key: "1234568", secret: "my_secret" }
+        )
+      }
+      let(:data) {
+        {
+          meta: { responseCode: 400, limit: 50, offset: 0, message: "Invalid merchant team." },
+          data: []
+        }.deep_stringify_keys
+      }
+
+      before do
+        allow(vine_api_service).to receive(:voucher_validation).and_return(
+          mock_api_response(success: false, status: 400, data: )
+        )
+      end
+
+      it "returns nil" do
+        expect(validate_voucher_service.validate).to be_nil
+      end
+
+      it "adds an error message" do
+        validate_voucher_service.validate
+
+        expect(validate_voucher_service.errors).to include(
+          { invalid_voucher: "The voucher is not valid" }
+        )
+      end
+
+      it "doesn't creates a new VINE voucher" do
+        expect { validate_voucher_service.validate }.not_to change { Voucher.count }
+      end
+    end
+
+    context "when creating a new voucher fails" do
+      let!(:vine_connected_app) {
+        ConnectedApps::Vine.create(
+          enterprise: distributor, data: { api_key: "1234568", secret: "my_secret" }
+        )
+      }
+      let(:data) {
+        {
+          meta: { responseCode: 200, limit: 50, offset: 0, message: "" },
+          data: {
+            id: "9d2437c8-4559-4dda-802e-8d9c642a0c1d",
+            voucher_short_code: voucher_code,
+            voucher_set_id: "9d24349c-1fe8-4090-988b-d7355ed32559",
+            is_test: 1,
+            voucher_value_original: 500,
+            voucher_value_remaining: '',
+            num_voucher_redemptions: 0,
+            last_redemption_at: "null",
+            created_at: "2024-10-01T13:20:02.000000Z",
+            updated_at: "2024-10-01T13:20:02.000000Z",
+            deleted_at: "null"
+          }
+        }.deep_stringify_keys
+      }
+
+      before do
+        allow(vine_api_service).to receive(:voucher_validation).and_return(
+          mock_api_response(success: true, status: 200, data: )
+        )
+      end
+
+      it "returns an invalid voucher" do
+        voucher = validate_voucher_service.validate
+        expect(voucher).not_to be_valid
```

Comment:
We can also specify what's not valid. Something like this?
```suggestion
        expect(voucher).not_to be_valid
        expect(voucher.errors[:amount]).to eq "Amount cannot be blank"
```